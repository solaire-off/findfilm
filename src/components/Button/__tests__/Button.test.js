import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { Button } from "..";

it("Current render with custom style and size classes", () => {
  const { asFragment } = render(
    <Button
      type="button"
      buttonStyle="btn--outline-primary"
      buttonSize="btn--lg"
      additionalClass="custom-btn-class"
    >
      + Add Movie
    </Button>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Current render with default style and size", () => {
  const { asFragment } = render(
    <Button buttonStyle="btn--wrong-style" buttonSize="btn--wrong-size">
      + Add Movie
    </Button>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Callback onClick work", () => {
  const mockCallback = jest.fn();
  const buttonText = "+ Add Movie";
  render(
    <Button
      type="button"
      onClick={mockCallback}
      buttonStyle="btn--outline-primary"
    >
      {buttonText}
    </Button>
  );

  userEvent.click(screen.getByText(buttonText));
  expect(mockCallback).toBeCalledTimes(1);
});
