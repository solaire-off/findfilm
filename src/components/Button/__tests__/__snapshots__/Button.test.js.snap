// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current render with custom style and size classes 1`] = `
<DocumentFragment>
  <button
    class="btn btn--outline-primary btn--lg custom-btn-class"
    type="button"
  >
    + Add Movie
  </button>
</DocumentFragment>
`;

exports[`Current render with default style and size 1`] = `
<DocumentFragment>
  <button
    class="btn btn--primary  "
    type="button"
  >
    + Add Movie
  </button>
</DocumentFragment>
`;
